src/middleware.ts

```diff
@@ -15,8 +15,10 @@ export const hashSecret = async (secret: string) => {
 };
 
 export const datocms = defineMiddleware(async ({ locals }, next) => {
-  locals.datocmsEnvironment = datocmsEnvironment;
-  locals.datocmsToken = getSecret('DATOCMS_READONLY_API_TOKEN');
+  Object.assign(locals, {
```

Comment:
From the Astro docs. Should set new locals through `Object.assign()`

Reply:
I see that it is intended as being immutable in page and middleware context. 

> Astro 5.0 requires you to append values to the existing locals object without deleting it. Locals in middleware, API endpoints, and pages, can no longer be completely overridden.
https://5-0-0-beta.docs.astro.build/en/guides/upgrade-to/v5/#changed-adding-values-to-contextlocals

Reply:
`datocmsEnvironment: datocmsEnvironment,` can be simplified to `datocmsEnvironment`.